src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -397,6 +397,17 @@ internal Symbol BindNamespaceOrTypeOrAliasSymbol(ExpressionSyntax syntax, Diagno
 
         private TypeSymbol BindTupleType(TupleTypeSyntax syntax, DiagnosticBag diagnostics)
         {
+
+            // If the tuple type is bound in a declaration context then we must have the
+            // TupleElementNamesAttribute to emit
+            if (syntax.IsTypeInContextWhichNeedsTupleNamesAttribute() &&
```

Comment:
If we do not encode names of tuples that do not have any named fields, perhaps this could be not an error for cases like   `(int, int)`  - i.e. you would not need the attribute as long as you do not have named fields.